test/modules/browser/esm-browser.test.ts

```diff
@@ -0,0 +1,92 @@
+import * as path from 'path'
+import { invariant } from 'outvariant'
+import { createTeardown } from 'fs-teardown'
+import * as express from 'express'
+import { HttpServer } from '@open-draft/test-server/http'
+import { test, expect } from '@playwright/test'
+import { spyOnConsole } from 'page-with'
+import { startDevServer } from '@web/dev-server'
+import { installLibrary } from '../module-utils'
+
+type DevServer = Awaited<ReturnType<typeof startDevServer>>
+
+const fsMock = createTeardown({
+  rootDir: path.resolve(__dirname, 'esm-browser-tests'),
+  paths: {
+    'package.json': JSON.stringify({ type: 'module' }),
+  },
+})
+
+let devServer: DevServer
+
+function getDevServerUrl(): string {
+  const address = devServer.server.address()
+
+  invariant(address, 'Failed to retrieve dev server url: null')
+
+  if (typeof address === 'string') {
+    return new URL(address).href
+  }
+
+  return new URL(`http://localhost:${address.port}`).href
+}
+
+const httpServer = new HttpServer((app) => {
+  app.use(express.static(fsMock.resolve('.')))
+})
+
+test.beforeAll(async () => {
+  devServer = await startDevServer({
+    config: {
+      rootDir: fsMock.resolve('.'),
+      port: 0,
+      nodeResolve: {
+        exportConditions: ['browser'],
```

Comment:
I don't think we even need the exportCondition here anymore btw, it should just work with `nodeResolve: true`

Reply:
@thepassle, good point!